aws/rust-runtime/aws-types/src/credentials/credentials_impl.rs

```diff
@@ -71,6 +76,36 @@ impl Credentials {
         }))
     }
 
+    /// Creates `Credentials` from hardcoded access key, secret key, and session token.
+    ///
+    /// _Note: In general, you should prefer to use the credential providers that come
+    /// with the AWS SDK to get credentials. It is __NOT__ secure to hardcode credentials
+    /// into your application. Only use this function if you really know what you're doing._
```

Comment:
not sure about "if you know what you're doing" is the right sentiment. We should also probably suggest that people use `Credentials::new` if they're writing their own credential provider